### PR TITLE
[Simplified login] initial screen for Jeptack installation

### LINKED ISSUE #7847: [Simplified login] Initial error screen for native Jetpack installation
Initial error screen to start the installation/connection flow:

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #7847 

### Description
This PR implements the initial screen of native Jetpack installation/connection flow.
Even though the PR seems big, it's just because it adds a new screen with its boilerplate.

ℹ️ I'm still waiting on the designer's confirmation on the usage of the illustrations for dark mode, but I don't think this should block the PR, we can always update them separately.

⚠️ Please don't merge before #7795 

### Testing instructions
1. Make sure you force the usage of the native Jetpack installation (either by adding your device to the list of test devices in the AB test in Firebase Console, or by hardcoding the value in `JetpackInstallationExperiment`)
2. Open the app, then sign in using your WordPress.com account.
3. In the site picker, click on Add Store, then choose "Connect an existing store".
4. Continue the steps of the following test cases:

###### Jetpack not installed flow
5. Enter the address of a site that doesn't have Jetpack.
6. Click on Continue
7. Confirm that the screen matching the Jetpack installation is shown.

##### Jetpack installed
5. Enter the address of a site that has Jetpack.
6. Click on Continue.
7. Confirm that the screen matching the Jetpack connection is shown.

### Images/gif
| | light | dark |
| --- | --- | --- |
| Install | image | image |
| Connect | image | image |

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/sitediscovery/SitePickerSiteDiscoveryViewModel.kt [hunk 1]
```diff
@@ -162,7 +163,12 @@ class SitePickerSiteDiscoveryViewModel @Inject constructor(
     }
 
     private suspend fun startSiteDiscovery() {
-        sitePickRepository.fetchSiteInfo(siteAddressFlow.value).fold(
+        val siteAddress = siteAddressFlow.value
+        if (sitePickRepository.getSiteBySiteUrl(siteAddress) != null) {
+            navigateBackToSitePicker()
+            return
+        }
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/sitediscovery/SitePickerSiteDiscoveryViewModel.kt:170]: This handles an unlikely edge case, the user enters the address of a site that's already connected to their store, and that was present in the site picker before, in this case, we'll return to the site picker while selecting this store to auto-login to it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/sitediscovery/SitePickerSiteDiscoveryViewModel.kt [hunk 2]
```diff
@@ -178,14 +184,40 @@ class SitePickerSiteDiscoveryViewModel @Inject constructor(
...
+                                    siteAddress = siteAddress,
+                                    isJetpackInstalled = false,
+                                    isJetpackConnected = false
+                                )
+                            )
+                        } else {
+                            stepFlow.value = Step.JetpackUnavailable
+                        }
+                    }
+                    !it.isWPCom -> {
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/sitediscovery/SitePickerSiteDiscoveryViewModel.kt:206]: At this step, even if Jetpack is connected, we are sure that the site is not connected to the user's account, so we'll continue to the JetpackActivation flow.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/TextExts.kt [hunk 3]
```diff
@@ -26,7 +26,7 @@ const val URL_ANNOTATION_TAG = "url"
  */
 @Composable
 fun annotatedStringRes(@StringRes stringResId: Int, vararg args: Any): AnnotatedString {
-    val string = stringResource(id = stringResId, args)
+    val string = stringResource(id = stringResId, *args)
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/TextExts.kt:29]: The varargs usage here was broken, this fixes it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchErrorScreen.kt [hunk 4]
```diff
@@ -222,7 +222,10 @@ fun AccountMismatchErrorScreen(viewState: ViewState.MainState, modifier: Modifie
             val context = LocalContext.current
             ClickableText(
                 text = consent,
-                style = MaterialTheme.typography.caption.copy(textAlign = TextAlign.Center)
+                style = MaterialTheme.typography.caption.copy(
+                    textAlign = TextAlign.Center,
+                    color = MaterialTheme.colors.onSurface
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchErrorScreen.kt:227]: This is unrelated to this PR, but it fixes a bug that I noticed, the color was black on black on dark theme.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/sitediscovery/SitePickerSiteDiscoveryViewModel.kt [hunk 5]
```diff
@@ -162,6 +163,13 @@ class SitePickerSiteDiscoveryViewModel @Inject constructor(
     }
 
     private suspend fun startSiteDiscovery() {
+        // If the site is already connected to the account, go back to site picker
+        if (sitePickRepository.getSiteBySiteUrl(siteAddressFlow.value) != null) {
+            fetchedSiteUrl = siteAddressFlow.value
+            navigateBackToSitePicker()
+            return
+        }
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/sitediscovery/SitePickerSiteDiscoveryViewModel.kt:171]: This handles an unlikely edge case, the user enters the address of a site that's already connected to their store, and that was present in the site picker before, in this case, we'll return to the site picker while selecting this store to auto-login to it.

File: WooCommerce/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -290,6 +290,11 @@
     <string name="login_create_an_account">Create An Account</string>
     <string name="site_picker_create_new_store">Create a new store</string>
     <string name="site_picker_connect_existing_store">Connect an existing store</string>
+    <string name="login_jetpack_installation_screen_title">Connect Strore</string>
```
Comment (0nko) [inline WooCommerce/src/main/res/values/strings.xml]: Minor typo:
```suggestion
    <string name="login_jetpack_installation_screen_title">Connect Store</string>
```

**Review by 0nko — COMMENTED** [review]
Aside from a typo, everything is working as expected. Nice work!

File: WooCommerce/src/main/res/values/strings.xml (hunk 6 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/res/values/strings.xml]: Thanks for catching this 🙏.

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**